### DISCUSSION TITLE
Keybindings for nvim-cmp automatically added?

### DISCUSSION THREAD

**Discussion by firubat** [LazyVim/LazyVim#3744] [Q&A] [answered]
After updating neovim to 0.10 (and probably also LazyVim although not sure what version), a whole bunch of insert-mode keybindings for nvim-cmp appeared, those pointing to `cmp.utils.keymap.set_map` :
![image](https://github.com/LazyVim/LazyVim/assets/22095250/4fbbd586-ac8c-495d-bc52-5ce5557f5bd8)
and overridden some of my own keybindings.

I couldn't find out where they were actually defined (and what are they mapped to) because it's hidden behind this `cmp.utils.keymap.set_map` mapping (function?).

- Could anyone point me to the documentation of these bindings?
- Is there a way to make WhichKey show the actuall function of each bind?
- Is there a way to disable their creation? Preferably as a whole, not one-by-one.
- How can I override them? Setting a new keymap doesn't do that.

Thanks!

**Comment by firubat** [comment]
Can I bump this?

**Reply by folke** [reply]
they're probably just the defaults?
What's the problem here? 
To override check the lazy.nvim docs and cmp docs.

**Reply by firubat** [reply]
So the usual way of overriding by adding to `lua/config/keymaps.lua` another mapping doesn't work, I guess because that file is loaded before the plugin?

In `nvim-cmp`'s documentation I couldn't find an explicit explanation of how to override these keybindings. 

But eventually I found [this discussion](https://www.reddit.com/r/neovim/comments/10r7l63/how_to_stop_nvimcmp_from_using_my_arrow_keys/) which gave the solution of overriding the entire default mapping presets, by passing an explicit table to `opts.mapping = cmp.mapping` instead of using `opts.mapping = cmp.mapping.preset.insert`.

**Comment by firubat** [comment, marked as answer]
Solution: redefine all of `nvim-cmp` keymaps. Template:
```
return {
  "hrsh7th/nvim-cmp",
   opts = function(_, opts)
    local cmp = require("cmp")
    opts.mapping = cmp.mapping({
      ["<KEY>"] = cmp.mapping.confirm({ select = false/true }),
      ["<KEY>"] = cmp.mapping.abort(),
      ["<KEY>"] = cmp.mapping.select_next_item(),
      ["<KEY>"] = cmp.mapping.select_prev_item(),
      ["<KEY>"] = cmp.mapping.scroll_docs(-4),
      ["<KEY>"] = cmp.mapping.scroll_docs(4),
    })
 end,
}
```
